onnxscript/function_libs/torch_aten/ops/nn.py

```diff
@@ -741,28 +741,104 @@ def aten_multilabel_margin_loss_forward(
     raise NotImplementedError()
 
 
+@torch_op("aten::nll_loss", trace_only=True)
 def aten_nll_loss(
-    self: TensorType,
-    target: TensorType,
-    weight: Optional[TensorType] = None,
+    self: TFloat,
+    target: Sequence[INT64],
+    weight: Optional[TFloat] = None,
     reduction: int = 1,
-    ignore_index: INT64 = -100,
-) -> TensorType:
-    """nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor"""
+    ignore_index: int = -100,
+) -> TFloat:
+    # nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor
+    reduction_vals = ["none", "mean", "sum"]
+    reduction_str = reduction_vals[reduction]
+
+    if weight is not None:
+        result = _aten_nll_loss_weight(
+            self, target, weight, reduction=reduction_str, ignore_index=ignore_index
+        )
+    else:
+        result = _aten_nll_loss_none(
+            self, target, reduction=reduction_str, ignore_index=ignore_index
+        )
+    return result
 
-    raise NotImplementedError()
 
+@torch_op("aten::nll_loss", overload=True)
+def _aten_nll_loss_weight(
+    self: TFloat,
+    target: Sequence[INT64],
+    weight: TFloat,
+    reduction: str,
+    ignore_index: int,
+) -> TFloat:
+    # nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor
+
+    rank_self = op.Size(op.Shape(self))
+    if rank_self == 1:
+        self = op.Unsqueeze(self, op.Constant(value_ints=[0]))
+        rank_target = op.Size(op.Shape(target))
```

Comment:
Not sure I understand the logic here. I assume target is meant to be a python list of ints, which will be mapped to a 1-dimensional tensor in ONNX. So, I am confused by the logic here.

Reply:
From onnx doc, the self must be at least 2-Dimension, and target must be at least 1-Dimension.
There have existing test case which self is 1-D data, and target is 0-D data.
So if self_rank=1, we need change it to 2. if rank_target is 0, we need change it to 1.